relay/relay.go

```diff
@@ -99,3 +99,15 @@ type Peer struct {
 	// to, which is also useful for stats.
 	Zone string
 }
+
+// RateLimitDropError is an error that a Relayer will get when it should drop
+// requests due to hitting a rate limit. This is kind of a hack, because rate
+// limiting of this nature isn't part of the actual TChannel protocol.
+//
+// When the Relayer gets an error like this when calling .Get() on an outbound
+// it will record that it has dropped the packet, but *won't* notify the client.
+type RateLimitDropError struct{}
```

Comment:
I don't see how that holds  here. All of the other errors that are used in this path are using the SystemError interface, but that appears to be only used for errors that correspond to TChannel protocol errors, i.e. things defined by the TChannel protocol. The TChannel protocol does not have a soft drop error.